falcon/request.py

```diff
@@ -256,6 +257,9 @@ def __init__(self, env, options=None):
         self.stream = env['wsgi.input']
         self.method = env['REQUEST_METHOD']
 
+        if self.stream.__class__.__name__ == InputWrapper.__name__:
```

Comment:
In this case, I'd recommend just replacing your changes to `request.py` and `request_helpers.py` with an addition of `InputWrapper` to L306. You can rebase and squash your commits to keep things clean. The additional tests you added in `test_wsgiref_inputwrapper_with_size.py` are still helpful, methinks.

By the way, [isinstance()](https://docs.python.org/2/library/functions.html#isinstance) supports passing a tuple of types to check, so that may be the most elegant way to go.

Thanks for working on this!